news-bundle/contao/modules/ModuleNews.php

```diff
@@ -80,19 +81,26 @@ protected function parseArticle($objArticle, $blnAddArchive=false, $strClass='',
 			$strClass = ' featured' . $strClass;
 		}
 
+		$url = $this->generateContentUrl($objArticle, $blnAddArchive);
+
 		$objTemplate->class = $strClass;
 		$objTemplate->newsHeadline = $objArticle->headline;
 		$objTemplate->subHeadline = $objArticle->subheadline;
 		$objTemplate->hasSubHeadline = $objArticle->subheadline ? true : false;
-		$objTemplate->linkHeadline = $this->generateLink($objArticle->headline, $objArticle, $blnAddArchive);
-		$objTemplate->more = $this->generateLink($objArticle->linkText ?: $GLOBALS['TL_LANG']['MSC']['more'], $objArticle, $blnAddArchive, true);
-		$objTemplate->link = $this->generateContentUrl($objArticle, $blnAddArchive);
+		$objTemplate->linkHeadline = $objArticle->headline;
 		$objTemplate->archive = $objArticle->getRelated('pid');
 		$objTemplate->count = $intCount; // see #5708
 		$objTemplate->text = '';
 		$objTemplate->hasTeaser = false;
 		$objTemplate->hasReader = true;
 
+		if (null !== $url)
+		{
+			$objTemplate->linkHeadline = $this->generateLink($objArticle->headline, $objArticle, $blnAddArchive);
```

Comment:
shouldn't the `generateLink` method handle the case of not having a link?

Reply:
I opted for not calling the method, if the generation already failed.

Reply:
Also I did not want to change the return type of the protected function.